str/trtools/merge_str_prep.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# pylint: disable=duplicate-code
+"""
+This script prepares GangSTR/EH VCF files for input into mergeSTR. 
+Required input: --caller, --input-dir, and external sample IDs
+For example: 
+analysis-runner --access-level test --dataset tob-wgs --description 'tester --output-dir 'tester' merge_prep.py --caller=eh --input-dir=gs://cpg-tob-wgs-main/str/expansionhunter/pure_repeats --dataset=tob-wgs TOBXXXX TOBXXXX
+
+Required packages: sample-metadata, hail, click, os
+pip install sample-metadata hail click
+
+"""
+import os
+import click
+import hailtop.batch as hb
+
+from sample_metadata.apis import SampleApi
+
+from cpg_utils.config import get_config
+from cpg_utils.hail_batch import remote_tmpdir, output_path
+
+config = get_config()
+
+REF_FASTA = 'gs://cpg-common-main/references/hg38/v0/Homo_sapiens_assembly38.fasta'
+BCFTOOLS_IMAGE = config['images']['bcftools']
+
+
+# inputs:
+# caller
+@click.option('--caller', help='gangstr or eh')
+# dataset
+@click.option('--dataset', help='dataset eg tob-wgs')
+# input directory
+@click.option('--input-dir', help='gs://...')
+# input sample ID
+@click.argument('external-wgs-ids', nargs=-1)
+@click.command()
+def main(
+    dataset, caller, input_dir, external_wgs_ids: list[str]
+):  # pylint: disable=missing-function-docstring
+
+    # Initializing Batch
+    backend = hb.ServiceBackend(
+        billing_project=get_config()['hail']['billing_project'],
+        remote_tmpdir=remote_tmpdir(),
+    )
+    b = hb.Batch(backend=backend, default_image=os.getenv('DRIVER_IMAGE'))
+
+    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
+        dataset, list(external_wgs_ids)
+    )
+
+    # Working with CRAM files requires the reference fasta
+    ref = b.read_input_group(
+        **dict(
+            base=REF_FASTA,
+            fai=REF_FASTA + '.fai',
+            dict=REF_FASTA.replace('.fasta', '').replace('.fna', '').replace('.fa', '')
+            + '.dict',
+        )
+    )
+
+    input_vcf_dict = {}
+
+    if caller == 'eh':
+        for id in list(external_id_to_cpg_id.values()):
+            input_vcf_dict[id] = input_dir + '/' + id + '_EH.vcf'
+    elif caller == 'gangstr':
+        for id in list(external_id_to_cpg_id.values()):
+            input_vcf_dict[id] = input_dir + '/' + id + '_gangstr.vcf'
+    else:
+        raise Exception('Invalid caller')
```

Comment:
```suggestion
    for id in list(external_id_to_cpg_id.values()):
        input_vcf_dict[id] = os.path.join(input_dir, f'{id}_{caller}.vcf')
```

This assumes the above change to make the `--caller` argument a strict choice, so you already know to expect one of `gangstr` or `eh`. Change to behaviour here is that `EH` is not capitalised.

Reply:
added to next commit